pkg/serviceaccounts/controllers/create_dockercfg_secrets.go

```diff
@@ -126,10 +126,15 @@ const (
 	maxSecretPrefixNameLength = maxNameLength - randomLength
 )
 
-func getDockercfgSecretNamePrefix(serviceAccount *api.ServiceAccount) string {
+func GetDockercfgSecretNamePrefix(serviceAccount *api.ServiceAccount) string {
 	return namer.GetName(serviceAccount.Name, "dockercfg-", maxSecretPrefixNameLength)
 }
-func getTokenSecretNamePrefix(serviceAccount *api.ServiceAccount) string {
+
+// GetTokenSecretNamePrefix creates the prefix used for the generated SA token secret. This is compatible with kube up until
```

Comment:
> not sure what "this" refers to

`GetTokenSecretNamePrefix` creates a name that is `saName-token-<random>` (if it fits) or `saName+hash-token-<random>` (if it doesn't fit).  The upstream code generates `saName-token-random` (if it fits) or `saName-random-random` if it doesn't fit.  I'd like to change the upstream code while refactoring this method out of this pacakge.